PR for SRVCOM-3678: Update the Serverless 1.36 release notes

kaldesai opened the pull request:
**Affected versions:** `serverless-docs-1.36`

**Tracking JIRA:** https://issues.redhat.com/browse/SRVCOM-3678

**Doc preview:**

- [Serverless 1.36 release notes](https://93729--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/serverless-release-notes#serverless-rn-1-36-0_serverless-release-notes)
- [Deprecated and removed features](https://93729--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/serverless-release-notes.html#serverless-deprecated-removed-features_serverless-release-notes)
- [Generally Available and Technology Preview features](https://93729--ocpdocs-pr.netlify.app/openshift-serverless/latest/about/serverless-release-notes.html#serverless-tech-preview-features_serverless-release-notes)

**Reviews:**
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->
- [x] SME has approved this change.
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Commit:
SRVCOM-3678-1.36-RNs: Updated the 1.36 release notes

kaldesai commented:
/label peer-review-needed

xenolinux commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,107 @@
+
+This issue is now fixed. The dispatcher correctly handles out-of-order event consumption during rebalancing and continues processing events without interruption.
+
+* Previously, a KafkaSource remained in a `Ready` state even when `KafkaSource.spec.net.tls.key` failed to load due to the use of unsupported TLS certificates in PKCS #1 format. This issue is now fixed. An appropriate error is now reported when attempting to create a KafkaBroker, KafkaChannel, KafkaSource, or KafkaSink using TLS certificates in an unsupported format.
```
Should the following instances be in back ticks? +1 to ticks if these are CR names
- KafkaBroker, KafkaChannel, KafkaSource, KafkaSink

kaldesai replied:
Yes, you are right! They should be in back ticks.

xenolinux commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,107 @@
+
+* The delivery of the Maven repository in the `devmode` image is not yet automated for offline environments.
+
+* When deploying a workflow in the `preview` profile on {ocp-product-title}, if the initial build fails and is later corrected, the Operator does not create the corresponding workflow deployment. As a result, the deployment remains missing and the SonataFlow status is not updated, even after the build is fixed.
```
```suggestion
* When deploying a workflow in the `preview` profile on {ocp-product-title}, if the initial build fails and is later corrected, the Operator does not create the corresponding workflow deployment. As a result, the deployment remains missing and the `SonataFlow` status is not updated, even after the build is fixed.
```

Should use the ticks right?

xenolinux reviewed (commented):
A few comments to consider; nicely done

maschmid commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,109 @@
+
+* Knative Eventing can now automatically discover and register EventTypes based on the structure of incoming events. This feature simplifies the configuration and management of EventTypes, reducing the need for manual definitions. This feature is available as a Technology Preview.
+
+* {ServerlessProductName} Eventing introduces EventTransform, a new API resource that allows you to declaratively transform JSON events without writing custom code. With EventTransform, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. EventTransform inte ...
```
EventTransform should also be Tech Preview (please update it here and create an entry in the TP table) 

(see the PM call notes)  CC: @matzew

matzew commented:
LGTM

ricardozanini reviewed (changes requested):
Could we have a breakdown of components in the release notes?

wmedvede commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,109 @@
+[id="known-issues-1-36-0_{context}"]
+== Known issues
+
+* The `openshift-serverless-1/logic-swf-builder-rhel8:1.35.0` image currently downloads persistence extensions from Maven during the build process. 
```
```suggestion
* The `openshift-serverless-1/logic-swf-builder-rhel8:1.35.0` and `openshift-serverless-1/logic-swf-builder-rhel8:1.36.0` images currently downloads persistence extensions from Maven during the build process. 
```

kaldesai commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,109 @@
+{ServerlessProductName} 1.36 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes:
+
+[id="new-features-1-36-0_{context}"]
+== New features
```
@ricardozanini I have made the changes accordingly. Thank you!

maschmid commented:
In the "Generally Available and Technology Preview features tracker" table, we list "Namespace-scoped brokers" as TP , while they are TP, they are also Deprecated, so perhaps we should drop them entirely from the "Generally Available and Technology Preview features tracker" table?

(On the Deprecated table, these are named "NamespacedKafka annotation" , but it's the same thing.  We should probably unify those, and just say "Namespace-scoped Kafka brokers" or something? CC @matzew

kaldesai commented:
@maschmid, Thank you so much for this suggestion. I have removed it from the "Generally Available and Technology Preview features" table and updated it to `Namespace-scoped Kafka brokers` in the "Deprecated and removed features" table.

maschmid commented:
QE approved

mburke5678 commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,125 @@
+
+* `IntegrationSource` and `IntegrationSink` are now available as a Technology Preview. These are Knative Eventing custom resources that support selected Kamelets from the Apache Camel project. Kamelets enables you to connect to third-party systems for improved connectivity, acting as either sources (event producers) or sinks (event consumers).
+
+* Knative Eventing can now automatically discover and register EventTypes based on the structure of incoming events. This feature simplifies the configuration and management of EventTypes, reducing the need for manual definitions. This feature is available as a Technology Preview.
```
According to the Knative docs, `EventType` is a Kubernetes object. 
```suggestion
* Knative Eventing can now automatically discover and register `EventTypes` objects based on the structure of incoming events. This feature simplifies the configuration and management of event types, reducing the need for manual definitions. This feature is available as a Technology Preview.
```

mburke5678 commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,125 @@
+
+* Knative Eventing can now automatically discover and register EventTypes based on the structure of incoming events. This feature simplifies the configuration and management of EventTypes, reducing the need for manual definitions. This feature is available as a Technology Preview.
+
+* {ServerlessProductName} Eventing introduces `EventTransform`, a new API resource that you can use to declaratively transform JSON events without writing custom code. With `EventTransform`, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. `EventTransfor ...
```
I see in the Knative docs that they use EventTransform without mark-up or _object_. I don't think we should, and should follow our standard 
```suggestion
* {ServerlessProductName} Eventing introduces `EventTransform` objects, a new API resource that you can use to declaratively transform JSON events without writing custom code. With `EventTransform` objects, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. Event transformations integrate seamlessly with Knative sources, triggers, and brokers, enhancing interoperability in event-driven architectures. This feature is now available as a Technology Preview.
```

mburke5678 commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,125 @@
+* {ServerlessProductName} Eventing introduces `EventTransform`, a new API resource that you can use to declaratively transform JSON events without writing custom code. With `EventTransform`, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. `EventTransfor ...
++
+
+See the following key features of `EventTransform`:
```
```suggestion
See the following key features of `EventTransform` objects:
```
or 
```suggestion
See the following key features of event transformations:
```

mburke5678 commented on modules/serverless-rn-1-36-0.adoc:
```diff
@@ -0,0 +1,125 @@
+[id="known-issues-osl-1-36-0_{context}"]
+=== {ServerlessLogicProductName}
+
+* If the `swf-dev-mode` image is started with a broken or invalid workflow definition, the container might enter a stuck state.
```
Is _stuck_ state the proper term?

mburke5678 commented:
@kaldesai I noted a few things that perhaps you could address in a follow-up PR, if you think it appropriate? Nothing that should block the merge. Thanks!
